enhancements/network/baremetal-networking.md

```diff
@@ -0,0 +1,260 @@
+
+#### coredns-mdns
+
+https://github.com/openshift/coredns-mdns/
+
+The `mdns` plugin for `coredns` was developed to perform DNS lookups
+based on discoverable information from mDNS.  This plugin will resolve both the
+`etcd-NNN` records, as well as the `_etcd-server-ssl._tcp.` SRV record.
+
+The list of `etcd` hosts included in the SRV record is based on the list of
+control plane nodes currently running.
+
+The IP addresses that the `etcd-NNN` host records resolve to comes from the
+mDNS advertisement sent out by the `mdns-publisher` on that control plane node.
+
+
+#### DNS Resolution During Bootstrapping
+
+One of the of virtual IP addresses required by in cluster networking infrastructure is used
+for our self-hosted internal DNS - the “DNS VIP”.  The location of the DNS VIP
+is [managed](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.conf.tmpl#L22)
+by `keepalived`, similar to the management of the API VIP.
+
+The control plane nodes are configured to use the DNS VIP as their primary DNS
+server.  The DNS VIP resides on the bootstrap host until the control plane
+nodes come up, and then it will move to one of the control plane nodes.
+
+`coredns-mdns` on the bootstrap node is [configured](https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-coredns-corefile.yaml)
+a little bit differently
+than `coredns-mdns` on the control plane nodes.  On the bootstrap node, it is
+[configured](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/coredns-corefile.tmpl)
+with a minimum number of hosts for the `etcd` SRV record based on
+the number of control plane nodes requested in `install-config.yaml`.
+
+This configuration is important.  It ensures that whenever `etcd` resolves this
+DNS SRV record to form the cluster, all of the expected control plane hosts are
+present in the results.
+
+#### DNS Resolution Post-Install
+
+After the bootstrap process completes, the DNS VIP moves to one of the control
+plane nodes.  All control plane nodes will continue to use this DNS VIP as
+their primary DNS server.
+
+If a control plane node is replaced, the contents of the DNS SRV record will
+automatically reflect the new node once it comes up and advertises itself via
+DNS.
+
+### Ingress High Availability
+
+There is a third VIP used by in cluster networking infrastructure, and that is for Ingress.
+The Ingress VIP will always reside on a node running an Ingress controller.
+This ensures that we provide high availability for ingress by default.
+
+The [configuration](https://github.com/openshift/machine-config-operator/blob/master/templates/worker/00-worker/baremetal/files/baremetal-keepalived-keepalived.yaml)
+of this mechanism used to determine which nodes are running an ingress controller
+is that `keepalived` will try to reach the local `haproxy` stats port number
+using `curl`.
```

Comment:
This only provides the dns and routing for the `default` ingresscontroller.. what about if the user configure multiple for sharding etc.

How would the user configure a vip setup for different one, can the user even do that. we should capture the limitations.

cc @russellb